src/imageio/imageio_jpegxl.c

```diff
@@ -169,6 +172,31 @@ dt_imageio_retval_t dt_imageio_open_jpegxl(dt_image_t *img, const char *filename
       continue;    // go to next loop iteration to process rest of the input
     }
 
+    if(status == JXL_DEC_BOX)
+    {
+      if(img->exif_inited) continue;  // There is no need for fallback reading of Exif data if Exiv2 has already done it for us
+
+      JxlBoxType type;
+
+      JxlDecoderReleaseBoxBuffer(decoder);	// Calling JxlDecoderReleaseBoxBuffer when no buffer is set is not an error
+
+      // Box decompression is not yet supported due to the lack of test images with brotli compressed Exif data.
+      status = JxlDecoderGetBoxType(decoder, type, JXL_FALSE);
+      if(status != JXL_DEC_SUCCESS) continue;
+
+      status = JxlDecoderGetBoxSizeRaw(decoder, &exif_size);
+      if((status != JXL_DEC_SUCCESS) || (exif_size <= 8)) continue;
+
+      if(memcmp(type, "Exif", 4) == 0)
+      {
+        // To get the Exif payload size we need to subtract 4 bytes of the box type FourCC
+        exif_size -= 4;
```

Comment:
It turns out this needs to be 8: https://github.com/libjxl/libjxl/issues/2022

Reply:
If we subtract only 4 when the correct number would be 8, then we will read extra 4 bytes for decoding as Exif. It doesn't cause any harm, I tested the code and it works correctly. If we keep it as it is now, nothing bad will happen.

You wrote "either the API documentation needs to be updated, or the implementation needs to change to match" and I still don't see what decision will be made by the libjxl project (they only set a milestone). If the implementation is changed to match the documentation (4 must be subtracted) and we subtract 8, we will not get the last 4 bytes of Exif data.

So, despite the possible incorrectness of this value, we should keep 4 for now.

Reply:
Ok, agreed there is no side effect. But please add a comment then and a link to the libjxl issue to revisit this perhaps.

Reply:
> But please add a comment then and a link to the libjxl issue to revisit this perhaps.

Good suggestion, I'll do.